### DISCUSSION TITLE
Cannot understand counters

### DISCUSSION THREAD

**Discussion by dml33** [photoprism/photoprism#1323] [Q&A] [answered]
Hi.
I am a newbie with photoprism. It looks awesome, so I am doing some tests before putting my collections in it.

I imported some photos and videos. In Originals folder there are 489 files as follows:

```
root@photoprism:/var/db/photoprism/storage/originals/2021/05 # ls *jpg | wc -l
     471
root@photoprism:/var/db/photoprism/storage/originals/2021/05 # ls *mp4 | wc -l
      18
```

In the app, I have the following:
![image](https://user-images.githubusercontent.com/25033130/119253264-b75e9780-bbb0-11eb-97a4-810bc666645b.png)
- Search -> 463
- Inside Search, only Stacks have contents -> no number is displayed, but there are 26. All of them of just 2 photos. Then, there should be 13 primary photos + 13 _no-primary_.
- Videos -> 18
- Places -> 3, but in the map there are two big bubbles: 449 and 14 -> 463. The bubbles contain the videos. No file is without location information (latitude, longitude).
![image](https://user-images.githubusercontent.com/25033130/119253373-45d31900-bbb1-11eb-891b-2447c21c2b6d.png)
- Library -> Originals -> 507. But, inside, it says "Folder contains 489 files". There is nothing under Hidden or Errors sections
![image](https://user-images.githubusercontent.com/25033130/119253494-c4c85180-bbb1-11eb-9a8a-01526c6ad9d7.png)

My question is why there is such incoherences:
- The original folder contains 489 files, not 507, as shown in the app in the Library section
- The Places do not show information about those 489 files, but just 463
- What "Search" shows? Because 463 (primaries) + 13 (no-primary) = 473, which is not 489

Thanks for the clarification!

**Comment by lastzero** [comment]
Files are actual files, photos are distinct photos / stacks. They may contain multiple files such as other formats, edited versions or XMP sidecar files. Thus the difference. Places counts distinct location labels like city names, not photos.

**Comment by dml33** [comment]
> Files are actual files,

In Library section it says 507, but when you go downwards and all the files are loaded, it says that there are 489 files, which is what the operating system says, too. So, the number _507_ must be wrong somehow.

> photos are distinct photos / stacks. They may contain multiple files such as other formats, edited versions or XMP sidecar files. Thus the difference. 

I have only edited some videos metadata, adding latitude and longitude, since it seems that the mobile does not record them in case of videos. 
Only Stack section shows some content: "26 results". Again, 463 + 13 (half of the stacked ones, since the primary photo is already counted) is 476, but not 489.

> Places counts distinct location labels like city names, not photos.

Do you mean a state or country name as a keyword or just a valid location, given by longitude and latitude coordinates? 

Thanks!

**Comment by lastzero** [comment]
> In Library section it says 507, but when you go downwards and all the files are loaded, it says that there are 489 files, which is what the operating system says, too. So, the number _507_ must be wrong somehow.

Did you count files in the *storage/sidecar/* folder as well?

> Do you mean a state or country name as a keyword or just a valid location, given by longitude and latitude coordinates?

Not numeric coordinates, actual location labels as you can see them e.g. in the cards view on https://demo.photoprism.org/browse

Example: *Chicago, Illinois, USA*

**Reply by dml33** [reply]
> > In Library section it says 507, but when you go downwards and all the files are loaded, it says that there are 489 files, which is what the operating system says, too. So, the number _507_ must be wrong somehow.
> 
> Did you count files in the _storage/sidecar/_ folder as well?

Total _yml_ files in sidecar folder:

```
root@photoprism:/var/db/photoprism/storage/sidecar/2021/05 # ls *yml | wc -l
     464
```

Why there are 464 _yml_ files instead of 463 (number in Search section)?

I notice that, in case of stacked photos, only the _yml_ file for the primary one is here. All _yml_ files have valid _Lat_ and _Lng_ fields.

Also, each mp4 has a jpg file associated and some of them have avc + jpg files.

Total number of files in sidecar folder:

```
root@Emby2:/var/db/photoprism/storage/sidecar/2021/05 # ls | wc -l
     490
```
The original folder has 489 files (total number of photos + videos):

```
root@Emby2:/var/db/photoprism/storage/originals/2021/05 # ls | wc -l
     489
```
> > Do you mean a state or country name as a keyword or just a valid location, given by longitude and latitude coordinates?
> 
> Not numeric coordinates, actual location labels as you can see them e.g. in the cards view on https://demo.photoprism.org/browse
> 
> Example: _Chicago, Illinois, USA_

Understood. 

All photos and videos have that kind of location labels. Since I set some location coordinates manually, I should tell the server to refresh or index that information again? 

Thanks

**Reply by lastzero** [reply]
> Why there are 464 _yml_ files instead of 463 (number in Search section)?

Maybe files were stacked after they were indexed initially? This may happen when you have more than one indexing worker e.g. on a multi-core system.

> All photos and videos have that kind of location labels. Since I set some location coordinates manually, I should tell the server to refresh or index that information again?

When you update metadata such as coordinates in files, make sure their file mod time is updated (some tools prevent this) so that PhotoPrism can see they were changed. Indexing should find the added / updated coordinates, unless there's conflicting information e.g. in related sidecar files.

**Reply by dml33** [reply]
I re-indexed everything again and now I have 472 _yml_ files. I did a diff with the former 464 files and the difference is because, for 8 stacked photos, I changed the primary photo. And now, due to this new indexing procedure, the _yml_ files for the new primary photos have been created.

Ok, I have understood that from the 489 original files, 463 stacked photos plus videos have been obtained.

But, still the 507 which appears in Library -> Originals section is unclear.

Thanks

**Reply by lastzero** [reply]
The number might include JPEG previews for RAWs or videos. Did you take that into account? Went through this with other users before and there was a plausible reason every time. If unsure, check the exact implementation in our source.

**Reply by lastzero** [reply]
Did you read our FAQs already?

**Reply by dml33** [reply]
Hi.
Yes, I read the FAQ. It says what are sidecar files and those things, but it does not say anything about how the total number of items in each section is calculated.  With the things you are telling me and other users, it could be completed.

You may understand that there are not many good options to organize photo collections and one of the greatest worries is to lose photos or that they remain hidden without any control about it. 

As I stated before, in my case, in Library -> Originals section, Photoprism says that 507 items are found.

But, after reindexing everything, in sidecar folder, there are only 498 files, including yml, jpg and avc files.

```
root@photoprism:/var/db/photoprism/storage/sidecar/2021/05 # ls | wc -l
     498
```
An brief example of its contents:
```
-rw-r--r--  1 photoprism  photoprism  207282197 May 22 20:48 20210521_191135_8A2284DA.mp4.avc
-rw-r--r--  1 photoprism  photoprism     398532 May 21 19:52 20210521_191135_8A2284DA.mp4.jpg
-rwxr-xr-x  1 photoprism  photoprism        489 May 23 13:13 20210521_191135_8A2284DA.yml
-rw-r--r--  1 photoprism  photoprism     612446 May 21 19:52 20210521_191401_A673D89C.mp4.jpg
-rwxr-xr-x  1 photoprism  photoprism        474 May 23 13:13 20210521_191401_A673D89C.yml
-rw-r--r--  1 photoprism  photoprism     659700 May 21 19:52 20210521_191404_28F5F72E.mp4.jpg
-rwxr-xr-x  1 photoprism  photoprism        474 May 23 13:13 20210521_191404_28F5F72E.yml
```

**Reply by lastzero** [reply]
You might want to run photoprism purge / cleanup in a terminal to be sure your index and folders are clean.

**Reply by lastzero** [reply]
Also make sure to reload the page in case it's a caching issue.

**Reply by lastzero** [reply]
The exact queries for the counters are here in the code:

https://github.com/photoprism/photoprism/blob/develop/internal/config/client.go#L308

**Reply by lastzero** [reply]
As stated here, the file browser counts are for files & folders combined:

https://github.com/photoprism/photoprism/blob/develop/internal/api/folder.go#L87

Hope this helps!

**Comment by graciousgrey** [comment, marked as answer]
We updated the FAQs on counts a while ago: https://docs.photoprism.org/user-guide/faq/#why-is-the-count-for-originals-higher-than-the-number-of-files-physically-existing-in-my-originals-directory
